src/PeriodicQuery.js

```diff
@@ -0,0 +1,58 @@
+const EventEmitter = require('events')
+const Stream = require('stream')
+
+const toArray = require('stream-to-array')
+
+module.exports = class PeriodicQuery extends EventEmitter {
+    constructor(queryFunction, retryInterval, retryTimeout) {
+        super()
+        this.queryFunction = queryFunction
+        this.retryInterval = retryInterval
+        this.retryTimeout = retryTimeout
+        this.readableStream = new Stream.Readable({
+            objectMode: true,
+            read() {},
+        })
+    }
+
+    async _startFetching() {
+        let results = await toArray(this.queryFunction())
```

Comment:
I'm left wondering whether it would be better to avoid copying the stream contents into an array before passing its contents on. It might lead to some momentarily heavy memory use at times.

In this situation, I don't think copying to an array is strictly necessary because it seems like what we really want to know here is whether the stream contains one or more data point(s) or not. Maybe something of the likes of:

```js
let dataSeen = false
const queryStream = this.queryFunction()
queryStream.on('data', (d) => this.readableStream.push(d))
queryStream.once('data', () => dataSeen = true)
queryStream.once('end', () => {
    if (dataSeen) {
        // do clean up stuff
    } else {
        // retry
    }
})
```